src/control/lib/daos/client/api.go

```diff
@@ -0,0 +1,289 @@
+package client
```

Comment:
Do you think it would be crossing too many boundaries to include the client API directly in the "daos" package? In terms of writing applications, users would end up having to import both daos and daos/client every time.